packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -323,34 +360,50 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
         (item) => !item.columnSpan || item.columnSpan === 1,
       );
 
-      const isFirstRow = heights.some((height) => height === 0);
-      const multiColumnItemWidth = parseInt(twoColumnItems[0].columnSpan, 10);
-      const availableSlotsOnFirstRow = heights.reduce(
-        (acc, height) => (height === 0 ? acc + 1 : acc),
-        0,
-      );
-      const fitsFirstRow = availableSlotsOnFirstRow >= multiColumnItemWidth + twoColumnIndex;
+      // The empty columns can be different from columnCount if there are
+      // items already positioned from previous batches
+      const emptyColumns = heights.reduce((acc, height) => (height === 0 ? acc + 1 : acc), 0);
+
+      const multiColumnItemColumnSpan = parseInt(twoColumnItems[0].columnSpan, 10);
+      const fitsFirstRow = emptyColumns >= multiColumnItemColumnSpan + twoColumnIndex;
 
-      // Skip the graph logic if the two column item batch is on the first line and fits
-      const skipGraph = isFirstRow && fitsFirstRow;
+      // Skip the graph logic if the two column item can be displayed on the first row
+      if (fitsFirstRow) {
+        const { positions: finalPositions, heights: finalHeights } = getSkipGraphPositions({
```

Comment:
i like that we're being explicit about the skipGraph case but i'm not sure this is better than the previous/current approach of just having an empty batch for the graph since this ultimately adds a bit more duplicated logic (e.g. updating the position and heights caches, getting the one column item positions, etc)

this is also something we'll likely need to tune a bit more in the future so rather than anchoring on `fitsFirstRow` I think keeping the current logic as-is makes more sense